vedacore/misc/checkpoint.py

```diff
@@ -281,6 +283,12 @@ def load_weights(model,
         state_dict = {k[7:]: v for k, v in state_dict.items()}
     if prefix is not None:
         state_dict = {'%s.%s' % (prefix, k): v for k, v in state_dict.items()}
+
+    if exclude is not None:
+        for k in list(state_dict.keys()):
+            if k.startswith(exclude):
+                del state_dict[k]
```

Comment:
Not too familiar/comfortable with the del keyword. @ReubenJ Can you also have a look?

Reply:
"del state_dict[k]" does exactly the same as "state_dict.pop(k)", so it removes the key-value pair with key "k" from the dictionary. Using del is just slightly more efficient than pop.

Reply:
Reads nicely as well :+1: